app/models/user.rb

```diff
@@ -343,6 +343,27 @@ def perform_export_photos!
     update exporting_photos: false
   end
 
+  ######### Data import ##################
+  mount_uploader :import, JsonUploader
+
+  def data_import(importedfile)
+    json_string = importedfile.read()
+    data_hash = JSON.parse(json_string)
+    if self.email != data_hash['user']['email']
+      self.unconfirmed_email = data_hash['user']['email']
+    end
+    self.show_community_spotlight_in_stream = data_hash['user']['show_community_spotlight_in_stream']
+    self.language = data_hash['user']['language']
+    self.disable_mail = data_hash['user']['disable_mail']
+    self.save
+    self.profile.update(bio: data_hash['user']['profile']['entity_data']['bio'])
+    self.profile.update(tag_string: data_hash['user']['profile']['entity_data']['tag_string'])
```

Comment:
- Style/RedundantSelf: Redundant `self` detected.
- Style/StringLiterals: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.